lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -234,6 +228,17 @@ func (ic *ICache) Add(imageConfig *metadata.ImageConfig) error {
 	ic.cacheByID[imageID] = imageConfig
 	ic.dirty = true
 
+	if imageConfig.Name == "" {
+		log.Debugf("Image %s has no name", imageID)
+		return nil
+	}
+	// construct reference after image is added into cacheByID, to make sure image without name can be added
```

Comment:
Per my understanding of our intention here, the comment below should suffice.

```
// Construct a reference after the image is added into cacheByID so that an image 
// without a name can at least be added to cacheByID.
```